### DISCUSSION TITLE
Multiple Stream with Docker - not working

### DISCUSSION THREAD

**Discussion by Snick32** [opendatacam/opendatacam#567] [Q&A]
Dear all,
i want to use two streams with a desktop machine and docker image.
My problem is, that i can not change the port. If i change in the yml File to 9000 and in the config.json also to 9000 it is not working. The odc doesent start. But with 8080 its working fine.
Is there a other value/place to change to the new port? And my other question is, how i add a second image in the yml File. 
This is my yml File:

  opendatacam:
    restart: always
    runtime: nvidia
    image: opendatacam/opendatacam:v3.0.2-desktop
    privileged: true
    volumes:
      - './config.json:/var/local/opendatacam/config.json'
    ports:
      - "8080:8080"
      - "8070:8070"
      - "8090:8090"
  opendatacam1:
    restart: always
    runtime: nvidia
    image: opendatacam/opendatacam:v3.0.2-desktop
    privileged: true
    volumes:
      - './config1.json:/var/local/opendatacam/config.json'
    ports:
      - "9000:9000"
      - "9071:9071"
      - "9091:9091"

Thanks for help!
Rico

**Comment by vsaw** [comment]
Take a look at the Docker Compose documentation: https://docs.docker.com/compose/compose-file/#ports

The first number is the Host port which can be changed, second port is the ODC container ports and must stay fixed. So try the following:

```
9080:8080
9070:8070
9090:8090
```

You may also need to change Service Names or similar to run multiple instances but I'm not a Docker expert.

**Comment by Snick32** [comment]
Dear Valentin,
thanks for the help. I tried and i created two docker yml Files. 

**Docker 1:**
version: "2.3"
services:
  opendatacam1:
    restart: always
    runtime: nvidia
    image: opendatacam/opendatacam:v3.0.2-desktop
    privileged: true
    volumes:
      - './config1.json:/var/local/opendatacam/config.json'
    ports:
      - "8080:8080"
      - "8070:8070"
      - "8090:8090"


  mongo:
    image: mongo
    restart: always
    ports:
      - "27017:27017"
    volumes:
      - mongodb_data_container:/data/db
volumes:
  mongodb_data_container:
  
**Docker 2:**
version: "2.3"
services:
  opendatacam:
    restart: always
    runtime: nvidia
    image: opendatacam/opendatacam:v3.0.2-desktop
    privileged: true
    volumes:
      - './config.json:/var/local/opendatacam/config.json'
    ports:
      - "9080:8080"
      - "9070:8070"
      - "9090:8090"


  mongo:
    image: mongo
    restart: always
    ports:
      - "27017:27017"
    volumes:
      - mongodb_data_container:/data/db
volumes:
  mongodb_data_container:


The ODC Server start twice on Port 9080 and 8080 succesfully. The first docker file runs perfect, but the second file (with the example mpg File) starts but stuck at the end of loading with no failure code.

opendatacam1_1  |  Allocate additional workspace_size = 52.43 MB
opendatacam1_1  | Loading weights from yolov4.weights...Demo
opendatacam1_1  | net.optimized_memory = 0
opendatacam1_1  | mini_batch = 1, batch = 8, time_steps = 1, train = 0
opendatacam1_1  | nms_kind: greedynms (1), beta = 0.600000
opendatacam1_1  | nms_kind: greedynms (1), beta = 0.600000
opendatacam1_1  | nms_kind: greedynms (1), beta = 0.600000
opendatacam1_1  | Done! Loaded 162 layers from weights-file
opendatacam1_1  | First request on console stream

Is there any Docker-Pro here the help me with the problem? 
Thx

**Reply by vsaw** [reply]
Not sure if this is the cause. But looking at the docker file it seems like you are starting two instances of MongoDB, both listening on the same port. This obviously fails and could be the reason why the second instance does not come up. 

So similar to how you changed names and ports for ODC, you should do the same for Mongo (or remove Mongo for the second file, this _should_ also work, but I've never tried it)
